spring-kafka-docs/src/main/antora/modules/ROOT/pages/kafka/kafka-queues.adoc

```diff
@@ -0,0 +1,274 @@
+            System.out.println("Received: " + record.value());
+        }
+    });
+
+    return container;
+}
+----
+
+[[share-container-properties]]
+=== Container Properties
+
+Share containers support a subset of the container properties available for regular consumers:
+
+* `topics`: Array of topic names to subscribe to
+* `groupId`: The share group ID
+* `clientId`: The client ID for the consumer
+* `kafkaConsumerProperties`: Additional consumer properties
+
+[NOTE]
+====
+Share consumers do not support:
+
+* Explicit partition assignment (`TopicPartitionOffset`)
+* Topic patterns
+* Manual offset management
+====
+
+[[share-annotation-driven-listeners]]
+== Annotation-Driven Listeners
+
+[[share-kafka-listener]]
+=== @KafkaListener with Share Consumers
+
+You can use `@KafkaListener` with share consumers by configuring a `ShareKafkaListenerContainerFactory`:
+
+[source,java]
+----
+@Configuration
+@EnableKafka
+public class ShareConsumerConfig {
+
+    @Bean
+    public ShareConsumerFactory<String, String> shareConsumerFactory() {
+        Map<String, Object> props = new HashMap<>();
+        props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
+        props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+        props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+        return new DefaultShareConsumerFactory<>(props);
+    }
+
+    @Bean
+    public ShareKafkaListenerContainerFactory<String, String> shareKafkaListenerContainerFactory(
+            ShareConsumerFactory<String, String> shareConsumerFactory) {
+        return new ShareKafkaListenerContainerFactory<>(shareConsumerFactory);
+    }
+}
+----
+
+Then use it in your listener:
+
+[source,java]
+----
+@Component
+public class ShareMessageListener {
+
+    @KafkaListener(
+        topics = "my-queue-topic",
+        containerFactory = "shareKafkaListenerContainerFactory",
+        groupId = "my-share-group"
+    )
+    public void listen(ConsumerRecord<String, String> record) {
+        System.out.println("Received from queue: " + record.value());
+        // Record is automatically acknowledged with ACCEPT
+    }
+}
+----
+
+[[share-group-configuration]]
+== Share Group Configuration
+
+Share groups require specific broker configuration to function properly.
+For testing with embedded Kafka, use:
+
+[source,java]
+----
+@EmbeddedKafka(
+    topics = {"my-queue-topic"},
+    brokerProperties = {
+        "unstable.api.versions.enable=true",
+        "group.coordinator.rebalance.protocols=classic,share",
+        "share.coordinator.state.topic.replication.factor=1",
+        "share.coordinator.state.topic.min.isr=1"
+    }
+)
+----
+
+[[share-group-offset-reset]]
+=== Share Group Offset Reset
+
+Unlike regular consumer groups, share groups use a different configuration for offset reset behavior.
+You can configure this programmatically:
```

Comment:
Is there a way to do this from configuration perspective, not programmatically ?

Reply:
I haven't really looked into it yet. Maybe we can revamp this are post M3?